File: HW14Assigned/low8/duck/RedHeadDuck.java
package low8.duck;
import low8.fly.*;
import low8.quack.*;
import low8.eggs.*;

public class RedHeadDuck extends Duck {
   public RedHeadDuck( ) {
      quackBehavior = new Quack( );
      flyBehavior = new FlyWithWings( );
      laysEggsBehavior = new LaysEggsBroody( );
   }

   public void display( ) {
      System.out.println("I’m a real redheaded duck");
   }
}

File: HW4Assigned/low8/RedHeadDuck.cpp
#include <iostream>
#include "RedHeadDuck.h"
#include "Quack.h"
#include "FlyWithWings.h"
#include "LaysEggsBroody.h"

RedHeadDuck::RedHeadDuck( ) {
   quackBehavior = new Quack( );
   flyBehavior = new FlyWithWings( );
   laysEggsBehavior = new LaysEggsBroody( );

}

void RedHeadDuck::display( ) {
   std::cout << "I’m a real redheaded duck" << std::endl;
}
